### PR TITLE
feat(api): inbox api filter preferences by workflow tags

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
Updated the Inbox API to allow filtering preferences by workflow tags.

### Screenshots

Screenshot 2024-09-17 at 13 59 52
Screenshot 2024-09-17 at 14 04 04

### CODE REVIEW

File: apps/api/src/app/inbox/usecases/get-preferences/get-preferences.usecase.ts [hunk 1]
```diff
@@ -43,8 +43,11 @@ export class GetPreferences {
     };
 
     const workflowList =
-      (await this.notificationTemplateRepository.getActiveList(command.organizationId, command.environmentId, true)) ||
-      [];
+      (await this.notificationTemplateRepository.filterActive({
+        organizationId: command.organizationId,
+        environmentId: command.environmentId,
+        tags: command.tags,
+      })) || [];
```
Comment (LetItRock) [inline apps/api/src/app/inbox/usecases/get-preferences/get-preferences.usecase.ts:50]: renamed this function and extended it to allow passing tags

File: apps/api/src/app/inbox/usecases/get-preferences/get-preferences.usecase.ts [hunk 2]
```diff
@@ -71,7 +74,7 @@ export class GetPreferences {
             id: workflow._id,
             identifier: workflow.triggers[0].identifier,
             name: workflow.name,
-            critical: workflow.critical || workflowPreference.template.critical,
+            critical: workflow.critical ?? workflowPreference.template.critical,
```
Comment (LetItRock) [inline apps/api/src/app/inbox/usecases/get-preferences/get-preferences.usecase.ts:77]: this was a bug here that was caught by the unit test, but we were not running them :/ 
basically `false || undefined` results in `undefined`

File: apps/api/src/app/inbox/e2e/get-preferences.e2e.ts [hunk 3]
```diff
@@ -49,4 +49,33 @@ describe('Get all preferences - /inbox/preferences (GET)', function () {
 
     expect(response.status).to.equal(401);
   });
+
+  it('should allow filtering preferences by tags', async function () {
```
Comment (LetItRock) [inline apps/api/src/app/inbox/e2e/get-preferences.e2e.ts:53]: unit test

File: apps/api/src/app/inbox/inbox.controller.ts [hunk 4]
```diff
@@ -103,13 +104,15 @@ export class InboxController {
   @UseGuards(AuthGuard('subscriberJwt'))
   @Get('/preferences')
   async getAllPreferences(
-    @SubscriberSession() subscriberSession: SubscriberEntity
+    @SubscriberSession() subscriberSession: SubscriberEntity,
+    @Query() query: GetPreferencesRequestDto
```
Comment (LetItRock) [inline apps/api/src/app/inbox/inbox.controller.ts:108]: the query params dto

File: apps/api/src/app/inbox/usecases/get-preferences/get-preferences.spec.ts [hunk 5]
```diff
@@ -169,4 +171,115 @@ describe('GetPreferences', () => {
 
     expect(result).to.deep.equal(mockedPreferencesResponse);
   });
+
+  it('it should return subscriber preferences filtered by tags', async () => {
```
Comment (LetItRock) [inline apps/api/src/app/inbox/usecases/get-preferences/get-preferences.spec.ts:175]: e2e test

**Review by SokratisVidros — APPROVED** [review]

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**